APIWrapper-Examples/BulkVRADeployment-Wrapper.ps1

```diff
@@ -0,0 +1,59 @@
+#Requires -PSEdition Core
+#Requires -Module ZertoApiWrapper
+
+#--------------------------------------------------------------------------------------------------------------#
+# Configure the variables below
+#--------------------------------------------------------------------------------------------------------------#
+
+$ESXiHostCSV = ".\BulkVRADeployment.csv"
+$ZertoServer = "GOES HERE"  # Enter the Hostname/IP of the Zerto Virtual Manager (ZVM) to connect to
+$ZertoPort = "9669" # Only Update if using a Nonstandard port.
+$ZertoUserName = "GOES HERE"    # Enter the ZVM username
+$ZertoPassword = "GOES HERE" | ConvertTo-SecureString -AsPlainText  # Enter the ZVM Password
+$ZertoCredentials = [PSCredential]::New($ZertoUserName, $ZertoPassword)
+
+#--------------------------------------------------------------------------------------------------------------#
+# Importing the CSV of ESXi hosts to deploy VRA to
+#--------------------------------------------------------------------------------------------------------------#
+
+$ESXiHostCSVImport = Import-Csv $ESXiHostCSV
+
+#--------------------------------------------------------------------------------------------------------------#
+#  Identify Unique Zerto Virtual Managers from CSV file
+#--------------------------------------------------------------------------------------------------------------#
+$ZVMName = $ESXiHostCSVImport | Select-Object ZVMName -Unique
+
+foreach ($ZVM in $ZVMName) {
+
+    # Establishes a connection to a ZVM
+    Connect-ZertoServer -zertoServer $ZVM.ZVMName -zertoPort $ZertoPort -credential $ZertoCredentials
+
+    #--------------------------------------------------------------------------------------------------------------#
+    #  Install Zerto VRA to each host specified in the imported CSV file
+    #--------------------------------------------------------------------------------------------------------------#
+
+    foreach ($ESXiHostName in $ESXiHostCSVImport) {
```

Comment:
From what I can tell, this will attempt to loop through every line in the CSV and attempt to install ALL VRAs rather than just the VRAs belonging to the ZVM that you are currently connected to. Maybe look at something like this...

```PowerShell
$HostVras = $ESXoHostCSVImport | Where-Object {$_.ZVMName -like $ZVM.ZVMName}
foreach ($Vra in $HostVras) {
...
}
```

Reply:
Changes look good. Thanks for updating